apps/src/templates/rubrics/LearningGoals.jsx

```diff
@@ -0,0 +1,392 @@
+import React, {useEffect, useState, useRef} from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
+import classnames from 'classnames';
+import style from './rubrics.module.scss';
+import {
+  learningGoalShape,
+  reportingDataShape,
+  studentLevelInfoShape,
+  submittedEvaluationShape,
+  aiEvaluationShape,
+} from './rubricShapes';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import {
+  BodyThreeText,
+  BodyFourText,
+  ExtraStrongText,
+  StrongText,
+  Heading6,
+} from '@cdo/apps/componentLibrary/typography';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import EvidenceLevels from './EvidenceLevels';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import AiAssessment from './AiAssessment';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import {UNDERSTANDING_LEVEL_STRINGS} from './rubricHelpers';
+
+const invalidUnderstanding = -1;
+
+export default function LearningGoals({
+  learningGoals,
+  teacherHasEnabledAi,
+  canProvideFeedback,
+  reportingData,
+  studentLevelInfo,
+  submittedEvaluation,
+  isStudent,
+  feedbackAdded,
+  setFeedbackAdded,
+  aiEvaluations,
+}) {
+  const [isAutosaving, setIsAutosaving] = useState(false);
+  const [autosaved, setAutosaved] = useState(false);
+  const [errorAutosaving, setErrorAutosaving] = useState(false);
```

Comment:
These three booleans look like they're supposed to be mutually exclusive, so they may be better captured by a single `autosaveStatus` state variable to avoid the complexity of keeping three separate booleans in sync properly.
```suggestion
  const STATUS = Object.freeze({ notStarted: 0, inProgress: 1, finished: 2, error: 3 });
  const [autosaveStatus, setAutosaveStatus] = useState(STATUS.notStarted);
```